packages/lesswrong/server/scripts/arbitalImport/arbitalImport.ts

```diff
@@ -2112,31 +2116,45 @@ async function importCoreTagAssignments(coreTagAssignmentsFile: string) {
       coreTagNames: row['Core Tag'].split(', '),
     };
   });
-  
-  const tagAssignmentIds = await Tags.find({ slug: {$in: tagAssignments.map(ta => ta.slug) }}).fetch();
-  const tagAssignmentIdsBySlug = Object.fromEntries(tagAssignmentIds.map(ta => [ta.slug, ta._id]));
-
-  const filteredTagAssignments = tagAssignments.filter(ta => ta.coreTagNames.every(tag => coreTagNames.has(tag)));
-
-  const tagAssignmentInserts: MongoBulkInsert<DbArbitalTagContentRel>[] = filteredTagAssignments.flatMap(ta => ta.coreTagNames.map(tag => ({
-    document: {
-      _id: randomId(),
-      parentDocumentId: coreTagIdsByName[tag],
-      childDocumentId: tagAssignmentIdsBySlug[ta.slug],
-      parentCollectionName: 'Tags',
-      childCollectionName: 'Tags',
-      type: 'parent-is-tag-of-child',
-      level: 0,
-      isStrong: true,
-      createdAt: new Date(),
-      legacyData: { coreTagAssignment: true },
-      schemaVersion: 1,
-    },
-  })));
 
-  console.log(`Inserting ${tagAssignmentInserts.length} tag assignments`);
+  const filteredTagAssignments = tagAssignments.filter(ta => {
+    const allCoreTagsExist = ta.coreTagNames.every(tag => coreTagNames.has(tag));
+    const hasSingleCoreTag = ta.coreTagNames.length === 1;
+    if (!hasSingleCoreTag) {
+      console.warn(
+        `Tag "${ta.slug}" has multiple core tags assigned (${ta.coreTagNames.join(', ')}). Only one core tag is supported per tag. Skipping this tag.`
+      );
+    }
+    return allCoreTagsExist && hasSingleCoreTag;
+  });
+
+  console.log(`Found ${filteredTagAssignments.length} valid tag assignments with a single core tag.`);
+
+  // Update each tag's coreTagId field
+  for (const ta of filteredTagAssignments) {
+    const tag = await Tags.findOne({ slug: ta.slug });
+    if (!tag) {
+      console.warn(`Tag with slug "${ta.slug}" not found. Skipping.`);
+      continue;
+    }
+
+    const coreTagName = ta.coreTagNames[0];
+    const coreTagId = coreTagIdsByName[coreTagName];
+    if (!coreTagId) {
+      console.warn(`Core tag "${coreTagName}" not found. Skipping.`);
+      continue;
+    }
+
+    // Update the tag's coreTagId field
+    await Tags.rawUpdateOne(
+      { _id: tag._id },
+      { $set: { coreTagId } }
+    );
+
+    console.log(`Assigned coreTagId "${coreTagId}" to tag "${tag.name}" (${tag.slug}).`);
+  }
```

Comment:
I'd wrap this in an `executePromiseQueue` of size... 10?  To speed it up when running against dev.